src/lib.rs

```diff
@@ -182,6 +172,21 @@ pub trait ToPublicKey: MiniscriptKey {
     /// Converts an object to a public key
     fn to_public_key(&self) -> bitcoin::PublicKey;
 
+    /// Convert an object to x-only pubkey
+    fn to_x_only_pubkey(&self) -> bitcoin::schnorr::PublicKey {
+        let pk = self.to_public_key();
+        bitcoin::schnorr::PublicKey::from(pk.key)
+    }
+
+    /// Serialize the key as bytes based on script context. Used when encoding miniscript into bitcoin script
+    fn push_to_builder<Ctx: ScriptContext>(&self, builder: script::Builder) -> script::Builder {
+        if Ctx::is_tapctx() {
```

Comment:
it might be nicer if the ScriptContext or Builder were more directly responsible for this logic than the MiniscriptKey.

Reply:
`Builder` belongs to rust-bitcoin and it would require an extra dummy trait to extend it here.

`ScriptContext` doesn't make a lot more sense to me than `MiniscriptKey`

Reply:
I can add a DummyTrait, one less method in `ToPublicKey`